File: UVA10050.java
import java.util.Scanner;

class UVA10050{
	static Scanner s = new Scanner(System.in);
	public static void main(String [] args){
		try{
			int numtimes = s.nextInt();
			while(numtimes>0){
				boolean[] days = new boolean[s.nextInt()+1];
				int[] hartal = new int[s.nextInt()];
				for(int i=0; i<hartal.length; i++){
					hartal[i] = s.nextInt();
				}
				for(int i=0; i<hartal.length; i++){
					for(int j=hartal[i]; j<days.length; j+=hartal[i]){
						days[j] = true;
					}
				}
				int counter =0;
				for(int i=1; i<days.length; i++){
					if(days[i]==true && i%7!=6 && i%7!=0){
						counter++;
					}
				}
				System.out.println(counter);
				numtimes--;
			}
		}
		catch(Exception e){}
	}
}

File: Competitive Programming 3/Chapter 2/1DArray/UVA10050.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

int main(){
	int TC, days, p;
	scanf("%d", &TC);
	while(TC--){
		scanf("%d", &days);
		scanf("%d", &p);
		int hartal[p];
		for(int i=0; i<p; i++){
			scanf("%d", &hartal[i]);
		}
		int counter =0;
		for(int i=1; i<=days; i++){
			if(i%7 == 6 || i%7==0){ continue;}
			for(int j=0; j<p; j++){
				if(i%hartal[j] ==0){
					counter++;
					break;
				}
			}
		}
		printf("%d\n", counter);

	}
    return 0;
}